geoutils/georaster.py

```diff
@@ -1771,3 +1772,77 @@ def split_bands(self: RasterType, copy: bool = False, subset: list[int] | int |
                 bands.append(raster)
 
         return bands
+
+    # Unfortunately, the Literal type doesn't work for py37, so a bool | None has to be used instead.
+    # When py38 is the least supported version, this should be 'as_frame: Literal[True]`) -> pd.GeoDataFrame etc.
+    @overload
+    def to_points(self, subset: float | int, as_frame: bool) -> gpd.GeoDataFrame:
+        ...
+
+    @overload
+    def to_points(self, subset: float | int, as_frame: None) -> np.ndarray:
+        ...
+
+    def to_points(self, subset: float | int = 1, as_frame: bool | None = None) -> np.ndarray:
+        """
+        Subset a point cloud of the raster.
+
+        If 'subset' is either 1 or is equal to the pixel count, all points are returned in order.
+        If 'subset' is smaller than 1 (for fractions) or the pixel count, a random sample is returned.
+
+        If the raster is not loaded, sampling will be done from disk without loading the entire Raster.
+
+        Formats:
+            * `as_frame` == None | False: A numpy ndarray of shape (N, 2 + nbands) with the columns [x, y, b1, b2..].
+            * `as_frame` == True: A GeoPandas GeoDataFrame with the columns ["b1", "b2", ..., "geometry"]
+
+        :param subset: The point count or fraction. If 'subset' > 1, it's parsed as a count.
+        :param as_frame: Return a GeoDataFrame with a geometry column and crs instead of an ndarray.
+
+        :raises ValueError: If the subset count or fraction is poorly formatted.
+
+        :returns: An ndarray/GeoDataFrame of the shape (N, 2 + nbands) where N is the subset count.
+        """
+        data_size = self.width * self.height
+
+        # Validate the subset argument.
+        if subset <= 0.0:
+            raise ValueError(f"Subset cannot be zero or negative (given value: {subset})")
+        # If the subset is equal to or less than 1, it is assumed to be a fraction.
+        if subset <= 1.0:
+            subset = int(data_size * subset)
+        else:
+            subset = int(subset)
+        if subset > data_size:
+            raise ValueError(f"Subset cannot exceed the size of the dataset ({subset} vs {data_size})")
+
+        # If the subset is smaller than the max size, take a random subset of indices, otherwise take the whole.
+        choice = np.random.randint(0, data_size - 1, subset) if subset != data_size else np.arange(data_size)
+
+        cols = choice % self.width
```

Comment:
I generally use `Raster.coords()` and subsample after flattening the array.
Here maybe you used `xy2ij` and `ij2xy` to account for the GDAL AREA or POINT attribute, thus better representing the location of the point data (left corner of pixel, or center of pixel)?! That is quite important actually!

Reply:
I first used `Raster.coords()` but this allocates two arrays for the whole dataset. Potentially, the user may only want 10 or so samples, so allocating two full-sized arrays would be unnecessary to say the least!

I can expose the location argument, but it should definitely default to "center"

Reply:
Added in efe3ed5.